core/utils.py

```diff
@@ -7,16 +7,29 @@
 from core.models import UserProfile, Repository
 
 
+"""Generates the request URL based on the user details"""
+def get_request_url(username):
+
+	user_details_url = 'https://api.github.com/users/{}'.format(username)
+	response = requests.get(user_details_url)
+	if response.status_code == 200:
+		public_repos = response.json()['public_repos']
+		repos_url = 'https://api.github.com/users/{}/repos?per_page={}'.format(username,public_repos)
```

Comment:
It's a better approach but still it has two flaws:
1. You are making at least two network calls even if the number of repositories is very less.
2. Even if you do `per_page=1000` GitHub will only send 100. That is the maximum. So, if a user has more than 100 repositories, this code will fail to grab it.